File: app/src/main/java/ru/frei/tasks/MainViewModel.java
package ru.frei.tasks;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import ru.frei.tasks.data.AppDatabase;
import ru.frei.tasks.data.TasksEntry;

public class MainViewModel extends AndroidViewModel {

    private LiveData<List<TasksEntry>> tasks;

    public MainViewModel(Application application) {
        super(application);
        AppDatabase database = AppDatabase.getInstance(this.getApplication());
        tasks = database.tasksDao().loadAllTasks();
    }

    public LiveData<List<TasksEntry>> getTasks() {
        return tasks;
    }
}

File: app/src/main/java/ru/frei/tasks/MainViewModel.kt
package ru.frei.tasks;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import ru.frei.tasks.data.AppDatabase;
import ru.frei.tasks.data.TasksEntry;

public class MainViewModel extends ViewModel {

    private LiveData<List<TasksEntry>> tasks;

    public MainViewModel(AppDatabase database, long listId) {
        tasks = database.tasksDao().loadAllTasksFrom(listId);
    }

    public LiveData<List<TasksEntry>> getTasks() {
        return tasks;
    }
}
